lib/improver/percentile.py

```diff
@@ -0,0 +1,149 @@
+            DEFAULT_PERCENTILES.
+
+        """
+        if not isinstance(collapse_coord, list):
+            collapse_coord = [collapse_coord]
+        for coord_item in collapse_coord:
+            _type_test(coord_item)
+
+        if percentiles is not None:
+            self.percentiles = [int(value) for value in percentiles]
+        else:
+            self.percentiles = self.DEFAULT_PERCENTILES
+
+        # Collapsing multiple coordinates results in a new percentile
+        # coordinate, its name suffixed by the original coordinate names. Such
+        # a collapse is cummutative (i.e. coordinate order doesn't matter).
+        # However the coordinates are sorted such that the resulting
+        # percentile coordinate has a consistent name regardless of the order
+        # in which the user provides the original coordinate names.
+        self.collapse_coord = sorted(collapse_coord, _sort_coord)
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        desc = ('<PercentileConverter: collapse_coord={}, percentiles={}'
+                .format(self.collapse_coord, self.percentiles))
+        return desc
+
+    def process(self, cube):
+        """Create a cube containing the percentiles as a new dimension.
+
+        What's generated is:
+            * 13 percentiles - (5%, 10%, 20%, 25%, 30%, 40%, 50%, 60%,
+              70%, 75%, 80%, 90%, 95%)
+
+        Parameters
+        ----------
+        cube : iris.cube.Cube instance
+            Given the collapse coordinate, convert the set of values
+            along that coordinate into a PDF and extract percentiles
+            and min, max, mean, stdev.
+
+        Returns
+        -------
+        cube : iris.cube.Cube instance
+            A single merged cube of all the cubes produced by each percentile
+            collapse.
+
+        """
+        # Store data type and enforce the same type on return.
+        data_type = cube.dtype
+
+        try:
+            result = cube.collapsed(self.collapse_coord,
+                                    iris.analysis.PERCENTILE,
+                                    percent=self.percentiles)
+            result.data = result.data.astype(data_type)
+            return result
+        except:
+            raise CoordinateNotFoundError(
+                "Coordinate '{}' not found in cube passed to {}.".format(
+                    self.collapse_coord, self.__class__.__name__))
+
+
+def _type_test(collapse_coord):
+    """Test for input type."""
+    if not isinstance(collapse_coord, (basestring, iris.coords.Coord)):
+        raise ValueError('collapse_coord is {!r}, which does not specify '
+                         'a single coordinate.'.format(collapse_coord))
+
+
+def _sort_coord(item_a, item_b):
+    """Sorting function to alphabetically sort either strings representing
+    coordinates or iris.coords.Coord objects by their names.
+
```

Comment:
If this custom sort is to remain, a docstring is required. This should particularly describe how the `sorted` function is using the `_sort_coord` function, and why returning `1`, `0` or `-1` makes sense.